### ISSUE TITLE
NET 8 Blazor Cookie Authentication and Authorization Policy not the same as in NET 7

### ISSUE THREAD

**Issue by joecuevasjr** [dotnet/aspnetcore#52222]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Describe the bug

I've got a NET 8 Blazor (Interactive Server - Per page/component interactivity) application  where Cookie Authentication has been added.
I handle cookie authentication via a minimal api endpoint, where I interact with the `HttpContext` and create a `ClaimsPrincipal` and call `context.SignInAsync`. Subsequently, a `Results.Redirect` call reloads the Blazor page where Authorization is now in effect.

The issue occurs when I add an authorization policy, such as:

````csharp
builder.Services.AddAuthorization(options =>
{
    options.AddPolicy("Age", policy =>
    {
        policy.RequireClaim("age", "21");
    });
});
````

If I add the following to a routable component, for example the `counter` page:
````csharp
@attribute [Authorize(Policy = "Age")]
````
And if the policy is not satisfied with the required claim, then the page redirects to the cookie's configured LoginPath.
However, I'm _already logged on_ and I expect the Routes.Razor `<NotAuthorized>` template directive to render in the Blazor component instead of a redirect to the default LoginPath `/Account/AccessDenied?ReturnUrl=`

### Expected Behavior

I'm expecting this to work as it did on NET 7. On Net 7, the authorized component correctly renders the `<NotAuthorized>` template directive when the policy based authorization requirement is not met. 

Here's my `Routes.razor` code:
````csharp
<CascadingAuthenticationState>
    <Router AppAssembly="@typeof(Program).Assembly">
        <Found Context="routeData">
            <AuthorizeRouteView RouteData="@routeData" DefaultLayout="@typeof(Layout.MainLayout)">
                <NotAuthorized>
                    <h1>Sorry</h1>
                    <p>You're not authorized to reach this page.</p>
                </NotAuthorized>
            </AuthorizeRouteView>
            <FocusOnNavigate RouteData="@routeData" Selector="h1" />
        </Found>
    </Router>
</CascadingAuthenticationState>
````

### .NET Version

8

**Comment by joecuevasjr** [comment]
An update to my issue. It seems to only affect Blazor 8 apps where interactivity is set to a "Per page/component" level. Once I set it to Global interactivity via `App.razor`'s `<HeadOutlet>` and `<Routes>` set to `@rendermode="@InteractiveServer"` it works as expected and the `<NotAuthorized>` template directive renders as expected and it no longer redirects to the cookie's default `LoginPath`. 🤔

**Comment by joecuevasjr** [comment]
I think I've got this figured out. I'll close this out with an explanation of my realization:

Since I'm trying to use Static Server Rendering mixed with Server Rendered components, the default for SSR in Routes.razor is to utilize the cookie LoginPath when I'm not authorized. I've come to the realization that my expectations were incorrect and the default case here is expected.

However, once authorized I was still having the issue of an Interactive Server component not properly receiving the cascading authorization state, but I was able to rectify that with a call to `builder.Services.AddCascadingAuthenticationState()` to support mixed render modes.

My original issue was that I was migrating a Blazor Server Net 7 app to a SSR/Interactive Server Blazor Net 8 app. Once My Net 8 was set to Global Interactive Server, everything was acting as is did in Net 7.
